src/hooks/bookmarks/use-all-bookmarks.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { ApiBookmark } from 'lib/learn-client/api/api-types'
+import { getAllBookmarks } from 'lib/learn-client/api/bookmark'
+import useAuthentication from 'hooks/use-authentication'
+
+interface UseAllBookmarksResult extends Omit<UseQueryResult, 'data'> {
+	bookmarks: undefined | ApiBookmark[]
```

Comment:
Would the data be 'undefined' if someone isn't authenticated? or what would cause that return

Reply:
The data is `undefined` while the data is being loaded for the first time. So in the case that someone is not authenticated, the query will never execute and the value will be `undefined`.